internal/manifest/list.go

```diff
@@ -65,9 +65,10 @@ type List interface {
 // ListUpdate includes the fields which a List's UpdateInstances() method will modify.
 // This is publicly visible as c/image/manifest.ListUpdate.
 type ListUpdate struct {
-	Digest    digest.Digest
-	Size      int64
-	MediaType string
+	Digest       digest.Digest
+	Size         int64
+	MediaType    string
+	ArtifactType string
```

Comment:
(1) AFAICS this breaks compatibility; previously `UpdateInstances` didn’t modify `ArtifactType`, now existing callers would clear it because they don’t set this field.

Arguably `artifactType` just wasn’t supported until very recently, so there are no existing callers that could have done the right thing, but there already is a bit of a pattern to this, and (2).

---

So I think the `ArtifactType` field should, for now, be added to `ListUpdate.ReadOnly` only. (Ideally, long term, we should have separate types for “data read from an instance” and “edit an instance with these updates”, but that’s not practical to do right now.)

That’s also consistent with the other fields we have added in recent months.

Reply:
Okay, changing it.